### ISSUE TITLE
install bats directly into the VM images

### ISSUE THREAD

**Issue by Luap99** [containers/automation_images#139]
```
VERSION=v1.1.0 ./hack/install_bats.sh
Cloning into '/tmp/tmp.WsaMSX61iB'...
fatal: unable to access 'https://github.com/bats-core/bats-core/': Failed to connect to github.com port 443: Connection timed out
make: *** [Makefile:842: .install.bats] Error 128
```
From one of my PRs: https://cirrus-ci.com/task/4733813895987200

To prevent flakes bats should be in the VM images and not installed at runtime.

bats should available in both fedora and debian repos,
on fedora v1.7.0 is available and on ubuntu (21.10/22.04) v1.2.1 https://packages.ubuntu.com/search?keywords=bats

Given that the CI script installs v1.1.0 for our testing using the packaged version should just work?

@edsantiago @cevich PTAL

**Comment by edsantiago** [comment]
We can't rely on system-provided bats in Ubuntu: https://bugs.launchpad.net/ubuntu/+source/bats/+bug/1882542

However, yes, obviously we should install it at VM-create time, never runtime. Network access at runtime must always be the absolute minimum.

**Comment by cevich** [comment]
Agreed, and [actually we already do this.](https://github.com/containers/automation_images/blob/main/cache_images/podman_tooling.sh#L31).  Though it's coming in from podman's `Makefile`, which...[ahh here's the bug.](https://github.com/containers/podman/blob/main/hack/install_bats.sh#L7)  I'm pretty sure that should be `if [[ "$(type -P bats)" != "" ]]; then`.  Also, updating the make target to specify the hard-coded `/usr/local/bin/bats` as a dependency will help avoid unnecessarily calling the script.  I can toss up a PR real quick...

**Comment by Luap99** [comment]
What version is installed in the VM? I think it would be great to upgrade to something more recent, if I remember correctly last time I used hack/get_ci_vm the bats version didn't have the `--filter` feature which is very hand when you only need to run a single test.

**Comment by cevich** [comment]
The bats package isn't installed in [fedora](https://github.com/containers/automation_images/blob/main/cache_images/fedora_packaging.sh) or [ubuntu](https://github.com/containers/automation_images/blob/main/cache_images/ubuntu_packaging.sh).  Instead, both clone the podman repo and call `make install.tools`.

Given @edsantiago (reasonable) objection to https://github.com/containers/podman/pull/14686 I'm thinking:
1. Remove the target and `hack/install_bats.sh` from podman.  Leave it up to developers to make sure the environment is sound (which also enables/supports testing of new/changed environments)
2. Here, add the `bats` package to Fedora VM setup script.
3. Add something like the old installer script into the Ubuntu VM setup script.

**Comment by edsantiago** [comment]
I don't know what you mean by "old installer script" in (3), but (1) and (2) sound perfect.

**Comment by cevich** [comment]
Oh I just mean something like the `install_bats.sh` we have in podman.
